sdks/python/apache_beam/pipeline.py

```diff
@@ -157,6 +157,113 @@ def _root_transform(self):
     """Returns the root transform of the transform stack."""
     return self.transforms_stack[0]
 
+  def _remove_labels_recursively(self, applied_transform):
+    for part in applied_transform.parts:
+      if part.full_label in self.applied_labels:
+        self.applied_labels.remove(part.full_label)
+      if part.parts:
+        for part2 in part.parts:
+          self._remove_labels_recursively(part2)
+
+  def _replace(self, override):
+
+    assert isinstance(override, PTransformOverride)
+    matcher = override.get_matcher()
+
+    output_map = {}
+
+    class OutputVisitor(PipelineVisitor): # pylint: disable=used-before-assignment
+
+      def __init__(self, pipeline, applied_labels):
+        self.pipeline = pipeline
+        self._applied_labels = applied_labels
```

Comment:
Yes, but that's done via self.pipeline._remove_labels_recursively, not via this attribute.